### ISSUE TITLE
Auto-update of Done & Stuck card dates

### ISSUE THREAD

**Issue by tayamueller** [ParabolInc/action#296]
**After #1212 and #1211 and before #213, and #612:**

## Issue - Enhancement

Cards naturally move, over time, from Future > Active >< Stuck > Done. To more accurately report the flow of work in Summaries and Archives, but also save our users from a bit of typing, automatically adjust the date of any card moved to Stuck or Done to [today's date]

### Acceptance Criteria (optional)
   - When a user moves a card to Stuck or Done, the due date changes to [today's date]
   - When a user moves card from Stuck or Done back to Active or Future, the date does not change unless changed manually.

- **Estimated effort:** 11 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by mattkrick** [comment]
closing in favor of #1000

**Comment by jordanh** [comment]
Stuck on: dates in data model

**Comment by jordanh** [comment]
Iceboxing for now

**Comment by ackernaut** [comment]
I think the target implementation looks something like card backs, card comments, and card event messages where the Due Date is the Due Date (doesn’t change when Done because now the meaning has changed and that is confusing) and there is a comment on the timeline that says something like “Terry moved this to Done — Tue, Jun 21 2018” on the timeline where comments and events are shown. See #2140 for more thoughts
